react/ProductSummaryList.tsx

```diff
@@ -130,6 +133,22 @@ function ProductSummaryList(props: PropsWithChildren<Props>) {
     preferredSKU,
   } = props
 
+  const [shippingOptions, setShippingOptions] = useState([])
+
+  const { getShippingOptionFromSession } = useSession()
+
+  useEffect(() => {
+    async function getShippingFromSession() {
+      const result = await getShippingOptionFromSession()
+
+      if (result && !equals(result, shippingOptions)) {
+        setShippingOptions(result)
+      }
+    }
+
+    getShippingFromSession()
+  }, [getShippingOptionFromSession, shippingOptions])
```

Comment:
```suggestion
  useEffect(() => {
    async function getShippingFromSession() {
      const result = await getShippingOptionFromSession()

      if (result) {
        setShippingOptions(result)
      }
    }

    getShippingFromSession()
  }, [getShippingOptionFromSession])
```
Wouldn't this achieve the same result?

When `!equals(result, shippingOptions)` is `true` the `shippingOption` will have the value from `result`
When `!equals(result, shippingOptions)` is `false` the `shippingOption` will also have the value from `result`

Reply:
this condition is to avoid a loop, since the function keeps updating the shippingOptions even when it's already the same value as result

Reply:
@thalytafabrine you can use the `setShippingOptions` callback function to do that. This is more performant.

```suggestion
  useEffect(() => {
    async function getShippingFromSession() {
      const result = await getShippingOptionFromSession()

      if (result) {
        setShippingOptions((currentShippingOptions) => equals(currentShippingOptions, result) ? currentShippingOptions : result)
      }
    }

    getShippingFromSession()
  }, [getShippingOptionFromSession])
```